packages/amplify-cli/src/init-steps/postInitSetup.ts

```diff
@@ -19,44 +24,46 @@ export async function postInitSetup(context: $TSContext) {
       await context.amplify.pushResources(context);
       await runPackage();
     } catch (e) {
-      if (e.name !== 'InvalidDirectiveError') {
-        context.print.error(`An error occurred during the push operation: ${e.message}`);
-      }
-      await context.usageData.emitError(e);
-      exitOnNextTick(1);
+      throw new AmplifyError('ProjectInitError', {
+        message: e.name === 'InvalidDirectiveError' ? 'Invalid directive' : 'An error occurred during the push operation',
+        details: e.message,
+        classification: 'InvalidProjectConfiguration',
```

Comment:
I don't think we can say for sure this is invalid project config at this point. This error could be something that went wrong on our end

Also, assuming InvalidDirectiveError is coming from the graphql transformer, that's not something that should be handled separately at this point in the code. I'd suggest we remove that if statement. If it needs some special handling it should happen closer to where it's thrown